test/emailpassword/config.test.js

```diff
@@ -42,4 +44,195 @@ describe(`configTest: ${printPath("[test/config.test.js]")}`, function () {
+                        formFields: [
+                            {
+                                id: "email",
+                            },
+                            {
+                                id: "password",
+                            },
+                        ],
+                    },
+                }),
+            ],
+        });
+
+        let emailpassword = await EmailPasswordRecipe.getInstanceOrThrowError();
+        assert(emailpassword.config.signUpFeature.formFields[0].validate !== undefined);
+        assert(emailpassword.config.signUpFeature.formFields[1].validate !== undefined);
+    });
+
+    it("test that giving optional true in email / password field should be ignored", async function () {
+        await startST();
+        STExpress.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [
+                EmailPassword.init({
+                    signUpFeature: {
+                        formFields: [
+                            {
+                                id: "email",
+                                optional: true,
+                            },
+                            {
+                                id: "password",
+                                optional: true,
+                            },
+                        ],
+                    },
+                }),
+            ],
+        });
+
+        let emailpassword = await EmailPasswordRecipe.getInstanceOrThrowError();
+        assert(!emailpassword.config.signUpFeature.formFields[0].optional);
+        assert(!emailpassword.config.signUpFeature.formFields[1].optional);
+    });
+
+    //Check that the default password and email validators work fine
+    it("test that default password and email validators work fine", async function () {
+        await startST();
+        STExpress.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [EmailPassword.init()],
+        });
+
+        let formFields = await EmailPasswordRecipe.getInstanceOrThrowError().config.signUpFeature.formFields;
+
+        let defaultEmailValidator = formFields.filter((f) => f.id === "email")[0].validate;
+        assert((await defaultEmailValidator("aaaaa")) === "Email is invalid");
+        assert((await defaultEmailValidator("aaaaaa@aaaaaa")) === "Email is invalid");
+        assert((await defaultEmailValidator("random  User   @randomMail.com")) === "Email is invalid");
+        try {
+            await defaultEmailValidator();
```

Comment:
I believe that this should return a string too now.. how is this test not failing?

This is a bug in the function. Please do not manipulate your tests to make them pass..